cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -335,6 +331,19 @@ def _add_resources(self):
                 head_node_alarms=self.head_node_alarms,
             )
 
+    def _add_internal_efs_shared_storage(self):
+        if self.config.head_node.shared_storage_efs_settings:
+            encrypted = self.config.head_node.shared_storage_efs_settings.encrypted
+        else:
+            encrypted = False
```

Comment:
[Single Responsibility / Maintainability] The resource `SharedStorageEfsSettings` should be the only responsible to determine the default value of the encryption. To this aim, you should define the default value into a constant `SharedStorageEfsSettings .DEFAULT_ENCRYPTION` and reference the constant here. In this way we do not risk to scatter default values in different places. This prevent scattering responsibilities in different places, makes the code easier to maintain as it avoids code duplications and risk of misalignments.